### DISCUSSION TITLE
PnP Search Results webpart Custom layout - getUnique helper.

### DISCUSSION THREAD

**Discussion by SSkj2013** [microsoft-search/pnp-modern-search#2914] [Q&A]
Version used:
4.8.0

Describe the bug:
{{getUnique}}/{{getUniqueCount}} are returning results page by page, and not for data.items as a whole. 

To Reproduce:
set up PnP Search Results webpart, and on page 2 of 4, select Custom Layout

Replace - {{getCountMessage @root.data.totalItemsCount @root.inputQueryText}} 
With - {{getCountMessage (getUniqueCount @root.data.items "MYCustomProperty") @root.inputQueryText}}

**Expected behavior**
My actual total is 1458 with duplicates on MyCustomProperty. The expectation is to filter out duplicates on MyCustomProperty, which are about 12 including the one that I want to show, and the expected result is 1447. 

**Screenshots** 
(with getTotalItemsCount)
<img width="218" alt="image" src="https://user-images.githubusercontent.com/95306710/232532572-894cbaef-0308-42d2-afa1-9d76ec143340.png">
when I replace with (getUniqueCount @root.data.items "MYCustomProperty") , the count is 10 per page, and there are a number of pages.
<img width="148" alt="image" src="https://user-images.githubusercontent.com/95306710/232533290-f653eb1b-b9ff-4bc4-bd6e-756079e7563b.png">

**Desktop (please complete the following information):**
 - Chrome, Edge, 

**Additional context**
Same behavior noticed with pnp search results v4.5.0

**Comment by wobba** [comment]
`getUnique` will only work across returned results on that page. 

What you want is support for [Collapse Specification](https://learn.microsoft.com/en-us/sharepoint/dev/general-development/customizing-search-results-in-sharepoint#collapse-similar-search-results-using-the-collapsespecification-property) which is not currently implemented. Also see #2088.

**Reply by SSkj2013** [reply]
@wobba Thank you! can collapse specification be worked in the Query Text area? or in the refinement filters section on the first page? I tried different things, but couldn't get it to work. 
![image](https://user-images.githubusercontent.com/95306710/232814665-4aa212d3-4c65-4521-a8e4-eb13ddb06f63.png)

**Reply by wobba** [reply]
No, we have to add a new web part setting for it.
